test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2028,3 +2028,97 @@ UCS_TEST_P(test_ucp_sockaddr_protocols_err, tag_rndv_unexp_put_scheme,
 }
 
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols_err)
+
+class test_ucp_sockaddr_protocols_err_sender
+      : public test_ucp_sockaddr_protocols {
+public:
+protected:
+    test_ucp_sockaddr_protocols_err_sender() {
+        set_tl_timeouts(m_env);
+        m_env.push_back(new ucs::scoped_setenv("UCX_IB_REG_METHODS",
+                                               "rcache,odp,direct"));
+    }
+
+    ucs::ptr_vector<ucs::scoped_setenv> m_env;
+};
+
+/* This test is quite tricky: is checks for incorrect behavior on RNDV send
+ * on DC transport: in case if sender EP was killed right after sent RTS
+ * then receiver may get incorrect/corrupted data */
+UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender, tag_rndv_killed_sender,
+           "RNDV_THRESH=10k", "RNDV_SCHEME=get_zcopy", "KEEPALIVE_INTERVAL=0",
+           "KEEPALIVE_NUM_EPS=inf")
+{
+    static size_t size = 64 * UCS_KBYTE;
+
+    /* Warmup */
+    test_tag_send_recv(size, true, false, false, false);
+
+    if (m_err_count > 0) {
+        UCS_TEST_SKIP_R("Failed to connect");
+    }
+
+    std::string send_buf(size, 'x');
+    std::string recv_buf(size, 'y');
+    std::string recv_copy(size, 'y');
+    std::string str_z(size, 'z');
+
+    void *rreq = NULL, *sreq = NULL;
+    int i;
+
+    rreq = ucp_tag_recv_nbx(receiver().worker(), &recv_buf[0], size, 0, 0,
+                            &ucp_request_null_param);
+    ASSERT_TRUE(UCS_PTR_IS_PTR(rreq));
+
+    sreq = ucp_tag_send_nbx(sender().ep(), &send_buf[0], size, 0,
+                            &ucp_request_null_param);
+    ASSERT_TRUE(UCS_PTR_IS_PTR(sreq));
+
+    for(i = 0; i < 1000; i++) {
+        /* allow sender to send RTS */
+        ucp_worker_progress(sender().worker());
+    }
```

Comment:
I'd re-write it to do unexpected TAG receiving, i.e.:
```suggestion
    sreq = ucp_tag_send_nbx(sender().ep(), &send_buf[0], size, 0,
                            &ucp_request_null_param);
    ASSERT_TRUE(UCS_PTR_IS_PTR(sreq));

    ucp_tag_message_h message;
    rreq = ucp_tag_msg_recv_nbx(receiver().worker(), &recv_buf[0], size,
                                message, )&ucp_request_null_param;
    request_wait(rreq);
```
1. do `ucp_tag_send_nbx()`
2. do `ucp_tag_msg_recv_nbx()`
3. kill sender
4. do `ucp_tag_recv_nbx()`

it ensures that RTS is 100% was received by a peer, but RNDV GET Zcopy hasn't yet started.